src/payment/strategies/barclaycard/barclaycard-payment-strategy.ts

```diff
@@ -0,0 +1,82 @@
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import * as paymentStatusTypes from '../../payment-status-types';
+import PaymentStrategy from '../payment-strategy';
+
+import { BrowserInfoRequest, DeviceOS } from './barclaycard';
+
+export default class BarclaycardPaymentStrategy implements PaymentStrategy {
+    constructor(
+        private _store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator
+    ) { }
+
+    async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment } = payload;
+
+        if (!payment) {
+            throw new PaymentArgumentInvalidError(['payment']);
+        }
+
+        const { paymentData } = payment;
+
+        await this._store.dispatch(this._orderActionCreator.submitOrder(payload, options));
+
+        return await this._store.dispatch(this._paymentActionCreator.initializeOffsitePayment(
+            payment.methodId,
+            payment.gatewayId,
+            {
+                ...paymentData,
+                formattedPayload: {
+                    browser_info: this._getBrowserInfo(),
```

Comment:
If we can define a standard contract for sending additional device information, we wouldn't need a specific strategy implementation for Barclaycard.